webuild-drafts/IntermediaryTesting.md

```diff
@@ -0,0 +1,538 @@
+1. **Request Augmentation:** The intermediary includes in the presentation request:
+  
+
+- User-friendly name of the actual RP
+  
+- Unique identifier of the actual RP
+  
+- URL of the Registrar for the RP
+  
+- Identifier of the intended use
+  
+
+2. **Online Verification:** The EUDIW contacts the Registrar (at the provided URL) in real-time to verify:
+  
+
+- The RP’s registration status
+  
+- The relationship between the RP
+  and intermediary
+  
+- Authorized attribute access
+  
+
+**Critical Gap:** At the moment of writing, no standardized interface specification exists for EUDIW-to-Registrar online verification. This represents a significant interoperability challenge.
+
+**Cross-Border Complexity:**
+
+- RPs must be registered in their home member state
+  
+- RPIs with international operations must support registration mechanisms from multiple jurisdictions
+  
+- Each member state may implement different registration approaches on top of the choice whether to use Registration Certificates or not
+  
+
+## 5.2       Embedded Disclosure Policy Challenge
+
+**Challenge:** Electronic Attestations of Attributes (EAAs) may include embedded disclosure policies that specify which types of RPs are authorized to receive the attributes. The wallet must correctly evaluate these policies against the actual RP, not the intermediary.
+
+**Context:** Annex III of CIR 2024/2979 [i.9] specifies how embedded disclosure policies allow attribute issuers can limit what relying parties are allowed to receive attribute of the attestation:
+
+1. No policy
+  
+2. only authenticated relying parties which are explicitly listed in the disclosure
+  policy.
+  
+3. only authenticated wallet-relying parties with wallet-relying party access certificates derived from a specific root (or list of specific roots) or intermediate certificate(s).
+  
+
+The wallet verifies whether the wallet-relying party complies with the requirements of the embedded disclosure policy and informs the wallet user of the result.
+
+NOTE: The last option is based on the access certificate, not on the registration certificate. Since in the case of an intermediary, the access certificate identifies the RP
+intermediary and not the RP, it is already clear that this is an issue. WE BUILD should try an alternative that is based on the Registration Certificate / online registrar identification, to propose a solution that can work in the case of an intermediary.
+
+**The RPI Complication:** If the wallet evaluates the disclosure policy based on the
+intermediary’s Access Certificate (which identifies the intermediary as, e.g.,
+an “authentication service provider”), it will produce incorrect warnings even
+when the actual RP is of an authorized type (e.g., included in the list in the
+embedded disclosure policy).
+
+**Correct Implementation Requirements:**
+
+- The EUDIW must extract the actual RP identity from either:
+  
+  - The Registration Certificate (linking RP to intermediary), or
```

Comment:
I am wondering about the value of having an intermediary if I therefore have to store the logs and all the signed artifacts by myself, provide my access certificate and problably also my registration certificates